### DISCUSSION TITLE
Are `+layout.server.ts` loads meant to rerun on every child route change?

### DISCUSSION THREAD

**Discussion by madeleineostoja** [sveltejs/kit#10386] [Q&A]
I have an issue that's driving me insane — our app has a `+layout.server.ts` route that initialises all user data. The only things it consumes are `depends` (for manual invalidation) and `locals` (for db client and auth session). But every time I navigate to any child route, the `load` function in it re-runs, and the client re-fetches the returned `__data.json`, taking about 400-800ms, which (on top of general SSR) makes my app feel very unresponsive when navigating between routes.

From the documentation on loading data, server load (vs universal load) should only rerun if one of its dependencies change. Since this layout doesn't use `url`, and uses a custom key in `depends()` for manual invalidation, I can't figure out why its constantly rerunning and blocking render.

I don't have any public facing code to share, but the pseudo-code for it is something like the following

```ts
export const load: LayoutServerLoad = async ({
  depends,
  locals: { db, getSession }
}) => {
  const session = await getSession();

  if (!session) {
    throw redirect(303, '/login');
  }

  depends('app:data');
  
  const {data, error} = await db.getData();
 
  if (error) {
    throw error(500, error.msg);
  }

  return data;
};
```

Anyone have any experience with this?

**Comment by madeleineostoja** [comment]
Anyone?

**Comment by geoffrich** [comment]
It shouldn't be re-running on every navigation, no. I've written similar layout loads and they've only rerun when expected. If you're able to repro in a separate project I'd be happy to take a look.

**Comment by madeleineostoja** [comment]
Thanks for the response! After a lot of debugging I finally traced it down to Sentry's Sveltekit integration, by default it instruments `load` with its own wrapper, which causes them to invalidate on every route change. I've filed an issue for it.

Sorry for the false alarm!
